src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This type provides means for instrumenting compiled methods for dynamic analysis.
+    /// It can be combined with other <see cref="Instrumenter"/>s.
+    /// </summary>
+    internal partial class DynamicAnalysisInjector : CompoundInstrumenter
```

Comment:
I'll defer to @AlekseyTs on this. I followed the naming convention he used for the sequence point injector.